src/IndexedDB.ts

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/* Simple wrapper around IndexedDB.
+ */
+
+let idb = null;
+
+async function idbInit(): Promise<void> {
+    idb = await new Promise((resolve, reject) => {
```

Comment:
should probably check for indexeddb support here (ie. check for window.indexedDB, and put that in a catch block because of firefox: https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/browser-index.js#L32)

Reply:
all the calls to the indexedDB functions are wrapped in `try` blocks now, so this should be OK.